ops/ccs-ops-misc/synthetic-data/scripts/synthea-automation/build_run_synthea.sh

```diff
@@ -0,0 +1,472 @@
+
+  # check the generated output file; must have more than just the header line
+  line_cnt=$(cat "${BFD_SYNTHEA_AUTO_LOCATION}/${BFD_CHARACTERISTICS_FILE_NAME}" |wc -l)
+  if [[ "$line_cnt" -gt 1 ]]; then
+    BFD_CHARACTERISTICS="${BFD_SYNTHEA_AUTO_LOCATION}/${BFD_CHARACTERISTICS_FILE_NAME}"
+    echo "${BFD_SYNTHEA_OUTPUT_LOCATION}/${BFD_CHARACTERISTICS_FILE_NAME} successfully discovered"
+  else
+    echo "failed to read meaningful data from: ${BFD_SYNTHEA_OUTPUT_LOCATION}/${BFD_CHARACTERISTICS_FILE_NAME}"
+  fi
+  deactivate
+}
+
+# Function to upload the new end state properties file to an S3 bucket; it also extracts
+# the newest bene_id_start variable for use in a later function/operation.
+upload_characteristics_file_to_s3(){
+  echo "upload ${BFD_CHARACTERISTICS_FILE_NAME} file to S3"
+  cd "${BFD_SYNTHEA_AUTO_LOCATION}"
+  source .venv/bin/activate
+  python3 ./s3_utilities.py "${BFD_SYNTHEA_AUTO_LOCATION}" "upload_characteristics"
+  deactivate
+}
+
+# Function to upload the new end state properties file to an S3 bucket; it also extracts
+# the newest bene_id_start variable for use in a later function/operation.
+upload_props_file_to_s3(){
+  echo "upload end_state.properties file to S3"
+  cd "${BFD_SYNTHEA_AUTO_LOCATION}"
+  source .venv/bin/activate
+  python3 ./s3_utilities.py "${BFD_SYNTHEA_OUTPUT_LOCATION}" "upload_prop" "${BFD_SYNTHEA_AUTO_LOCATION}"
+  deactivate
+}
+
+
+#----------------- GO! ------------------#
+# general fail-safe to perform cleanup of any directories and files germane to executing
+# this shell script. Commented out for now
+# clean_up
+
+# invoke function to clone the Synthea repo and build it.
+install_synthea_from_git
+
+# invoke function to clone the BFD repo.
+install_bfd_from_git
+
+# invoke function to create a python virtual environment.
+activate_py_env
+
+# invoke function to download proprietary Mitre mapping files.
+download_mapping_files_from_s3
+
+# invoke function to download proprietary Mitre shell script files.
+download_script_files_from_s3
+
+# invoke function to download (if available) a previous run's end_state.properties file.
+download_props_file_from_s3
+
+# invoke function to invoke BFD .py script that verifies that:
+#  1) we have all the files necessary to perform a synthea generation run.
+#  2) executes a synthea generation run
+prepare_and_run_synthea
+
+# Invoke a function to upload the generated RIF files to the appropriate BFD
+# ETL pipeline S3 /Incoming bucket, where the ETL process will pick them up and
+# load the synthetic data into database.
+upload_synthea_results_to_s3
+
+# Invoke a functionn to wait on / check the appropriate BFD ETL pipeline S3 bucket for
+# the 0_manifest file to appear in the S3 bucket's /Done folder.
+wait_for_manifest_done
+
+# Invoke a function that executes a .py script that performs validation of data for
+# the just executed ETL pipeline load.
+if ! $SKIP_VALIDATION; then
```

Comment:
skip validation should only skip the initial pre-validation of the previous run; we always want to validate the load we just loaded

Reply:
got it; fixed.